totest-manager.py

```diff
@@ -162,23 +173,49 @@ def overall_result(self, snapshot):
 
         number_of_fails = 0
         in_progress = False
-        machines = []
         for job in jobs:
-            machines.append(job['settings']['MACHINE'])
             # print json.dumps(job, sort_keys=True, indent=4)
             if job['result'] in ('failed', 'incomplete', 'skipped', 'user_cancelled', 'obsoleted'):
                 jobname = job['name']
-                # Record machines we have tests for
-                if jobname in self.known_failures:
-                    logger.debug("known failure %s ignored", jobname)
-                    self.known_failures.remove(jobname)
-                    continue
-                number_of_fails += 1
                 # print json.dumps(job, sort_keys=True, indent=4), jobname
-                failedmodule = self.find_failed_module(job['modules'])
                 url = 'https://openqa.opensuse.org/tests/%s' % job['id']
-                logger.info("job %s failed %s, see %s", jobname, 'early' if failedmodule is None else failedmodule, url)
-                # if number_of_fails < 3: continue
+                logger.info("job %s failed, see %s", jobname, url)
+                url = makeurl('https://openqa.opensuse.org',
+                              ['api', 'v1', 'jobs', str(job['id']), 'comments'])
+                f = self.api.retried_GET(url)
+                comments = json.load(f)
+                refs = set()
+                #pprint(comments)
+                labeled = 0
+                to_ignore = False
+                for comment in comments:
+                    for ref in comment['bugrefs']:
+                        refs.add(str(ref))
+                    if comment['userName'] == 'ttm' and comment['text'] == 'label:unknown_failure':
+                        labeled = comment['id']
+                    if comment['text'].find('@ttm ignore') >= 0:
+                        to_ignore = True
+                ignored = True
+                for ref in refs:
+                    if not ref in self.issues_to_ignore:
+                        if to_ignore:
+                            self.issues_to_ignore.append(ref)
+                            with open(ISSUE_FILE, 'a') as f:
```

Comment:
this is racy. there are several instances of ttm running. So you need locking or (better) use a separate file per project